### PR TITLE
Ncnews 11

### PR DESCRIPTION
**Pull request by MohamedYislam**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -263,6 +334,24 @@ describe('GET', () => {
                 expect(msg).toBe('topic not found');
             });
         })
+        test("can not be sorted by  items which are not columns in the table", () => {
+            return request(app)
+            .get('/api/articles?sortBy=username')
+            .expect(404)
```
Comment (MRombi) [inline __tests__/app.test.js]: We should expect a 400 instead of a 404

File: __tests__/app.test.js [hunk 2]
```diff
@@ -263,6 +334,24 @@ describe('GET', () => {
...
+            .then((response) => {
+                const { msg } = response.body;
+                expect(msg).toBe("can not sort by this critera");
+            });  
+        })
+        test("can not be sorted by  items which are not columns in the table", () => {
```
Comment (MRombi) [inline __tests__/app.test.js:361]: We should expect a 400 from a wrong Order

File: __tests__/app.test.js [hunk 3]
```diff
@@ -251,7 +299,30 @@ describe('GET', () => {
...
+                            author: expect.any(String),
+                            body: expect.any(String),
+                            created_at: expect.any(String),
+                            votes: expect.any(Number),
+                            count: expect.any(Number)
+                        })
+                    );
+                });
+            })
+        })
```
Comment (MRombi) [inline __tests__/app.test.js:331]: We should have a test for when the topic exists but has no articles, and we should expect an empty array

**Review by MRombi — COMMENTED** [review]
Interesting way to create the query, after actioning on the three issues I have highlighted you are free to merge 😄

File: __tests__/app.test.js (hunk 3 above)
Comment (MohamedYislam) [reply __tests__/app.test.js:331]: could you please expand how a topic can exist wtihout an article, since it is listed as not null in our database table

**Pull request merged**